decidim-core/vendor/assets/javascripts/form_datepicker.js.es6

```diff
@@ -20,7 +20,7 @@
         leftArrow: '<<',
         rightArrow: '>>'
       }).on('changeDate', (ev) => {
-        $(ev.target).siblings('input').val(exports.moment.utc(ev.date));
+        $(ev.target).siblings('input').val(exports.moment(ev.date));
```

Comment:
That's weird, we were having this issues because it was being sent in UTC, is your app configured in the correct timezone?